scripts/cfFindIP.sh

```diff
@@ -27,21 +42,52 @@ scriptDir=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
 resultDir="$scriptDir/../result"
 resultFile="$resultDir/$now-result.cf"
 
+#check if expected output folder exists and create if it's not availbe
+if [ ! -d "$resultDir" ]; then
+    mkdir -p "$resultDir"
+fi
+
+
 # Function fncCheckSubnet
 # Check Subnet
 function fncCheckSubnet {
-	local ipList resultFile 
+	local ipList resultFile timeoutCommand
 	ipList="$1"
 	resultFile="$2"
+# Check if 'timeout' command is available
+if command -v timeout >/dev/null 2>&1; then
+    timeoutCommand="timeout"
+else
+    # Check if 'gtimeout' command is available
+    if command -v gtimeout >/dev/null 2>&1; then
+        timeoutCommand="gtimeout"
+    else
+        # Check if 'perl-timeout' command is available
+        if command -v perl-timeout >/dev/null 2>&1; then
+            timeoutCommand="perl-timeout"
+        else
+            # Check if 'python-timeout' command is available
+            if command -v python-timeout >/dev/null 2>&1; then
+                timeoutCommand="python-timeout"
+            else
+                echo >&2 "I require 'timeout' command but it's not installed. Please install 'timeout' or an alternative command like 'gtimeout' or 'perl-timeout' or 'python-timeout' and try again."
+                exit 1
+            fi
+        fi
+    fi
+fi
+
+
 	for ip in ${ipList}
 	do
-		if timeout 1 bash -c "</dev/tcp/$ip/443" > /dev/null 2>&1;
-		then
-			timeMil=$(timeout 2 curl -s -w '%{time_total}\n' --resolve scan.sudoer.net:443:"$ip" https://scan.sudoer.net/data.100K --output /dev/null | xargs -I {} echo "{} * 1000 /1" | bc )
-			if [[ "$timeMil" ]]
+    	if $timeoutCommand 1 nc -z "$ip" 443; then #added nc to support alll comon distros 
+			timeMil=$($timeoutCommand 2 curl -s -w '%{time_total}\n' --resolve scan.sudoer.net:443:"$ip" https://scan.sudoer.net/data.100K --output /dev/null | xargs -I {} echo "{} * 1000 /1" | bc )
+			ping=$(ping -c 1 -W 1 "$ip" | tail -1| awk '{print $4}' | cut -d '/' -f 2) #adding ping time check
+
+			if [[ "$timeMil" && "$ping" ]] #considreing both ping and curl responses
 			then
-				echo "OK $ip"
-				echo "$ip" >> "$resultFile"
+				echo "OK $ip Latency: $timeMil PingTime: $ping" # adding more fields to the output
```

Comment:
Latency is not proper word
I believe response time is proper one

Reply:
changed in the latest commit